app/helpers/api_helper.rb

```diff
@@ -8,127 +8,94 @@
 ################################################################################
 
 module ApiHelper
+  include ApiInlineHelper
+
   def xml_boolean(xml, tag, val)
     str = val ? "true" : "false"
     xml.tag!(tag, type: "boolean", value: str)
-  rescue StandardError
   end
 
   def xml_integer(xml, tag, val)
-    str = begin
-            "%d" % val
-          rescue StandardError
-            ""
-          end
-    xml.tag!(tag, str, type: "integer")
-  rescue StandardError
+    val.present? &&
+      (str = val.try(:to_i)) &&
+      xml.tag!(tag, str.to_s, type: "integer")
```

Comment:
It would be easier for me to understand this if it were written as:
```
return false if val.blank?

str = val.try(:to_i)
xml.tag!(tag, str.to_s, type: "integer") if str
```
Essentially the same amount of code with less dependencies on understanding the parens and chaining.  Also gets rid of an assignment inside a statement which can be dangerous and confusing.  Should be just as efficient.

Reply:
Other similar cases below.

Reply:
Must be another case like the hated switch statement.  This in particular is a consequence of rubocop preventing what I'd really like to do:

  return if val.blank?
  return unless str = val.try(:to_i)
  xml.tag!(tag, str.to_s, type: "integer")

But rubocop requires a blank line in there that really grates on me.  (I like to keep small pieces of code together so that I can use [ and ] to move through the code one method at a time in vi.)

Since rubocop loves these screwy "this && that && do_something()" constructs, I thought I'd try that.  Alas, that I didn't realize this also is not acceptable!

Reply:
Ha!  I guess it all depends on what you like best and what your tools work best with.  I've gotten really used to that blank line after the last guard and I kind of like it now, but I can see it being a problem as well.  I'm ok with the code as written, I was just saying what would be more familiar and easier for me to read.